src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -341,6 +347,36 @@ private void OnPackageRemoved(Package package)
         }
 
         #endregion
+        internal bool CheckIfPackagesTargetOtherHosts(IEnumerable<PackageVersion> newPackageHeaders)
+        {
+            // determine if any of the packages are targeting other hosts
+            var containsPackagesThatTargetOtherHosts = false;
+
+            // Known hosts
+            var knownHosts = PackageManagerClient.GetKnownHosts();
+
+            // Sandbox, special case: Warn if any package targets only one known host
+            if (String.IsNullOrEmpty(Host))
+            {
+                containsPackagesThatTargetOtherHosts =
+                    newPackageHeaders.Any(y => y.host_dependencies != null && y.host_dependencies.Intersect(knownHosts).Count() == 1);
+            }
+            else
+            {
+                // Warn if there are packages targeting other hosts but not our host
+                var otherHosts = knownHosts.Except(new List<string>() { Host });
+                containsPackagesThatTargetOtherHosts = newPackageHeaders.Any(x =>
+                {
+                    // Is our host in the list?
+                    // If not, is any other host in the list?
+                    return x.host_dependencies != null && !x.host_dependencies.Contains(Host) && otherHosts.Any(y => x.host_dependencies != null && x.host_dependencies.Contains(y));
```

Comment:
Curious here, why do we need to check `otherHosts `? Can't we just check if our host is in the list and if not display the available list of hosts?

Reply:
I think his comment sums it up nicely:
`// Warn if there are packages targeting other hosts but not our host`

Reply:
I guess by doing this we can enable unloading these packages in the future at some point of Dynamo startup? It would be good to know the reason behind this for sure

Reply:
In:
```
return x.host_dependencies != null && !x.host_dependencies.Contains(Host) && otherHosts.Any(y => x.host_dependencies != null && x.host_dependencies.Contains(y));
```
why do we need to check `x.host_dependencies != null` twice?

Reply:
@aparajit-pratap good point, I think we can get rid of that check inside the `Any` statement.